app/src/components/SectionPicker.tsx

```diff
@@ -78,13 +95,25 @@ export default function SectionPicker(props: Props) {
                         "bg-gray-50 border border-gray-300 text-gray-900 text-sm rounded-lg p-2.5 mb-0.5"
                     }
                     placeholder={getSearchByStringExample(searchByString)}
-                    value={props.searchText}
-                    onChange={(e) => props.setSearchText(e.target.value)}
+                    value={searchText}
+                    onChange={(e) => setSearchText(e.target.value)}
                     autoComplete={"off"}
+                    onKeyDown={(e) => {
+                        if (e.key === "Enter") {
+                            searchCourse();
+                        }
+                    }}
                 />
+
+                <button
+                    onClick={searchCourse}
```

Comment:
Should be `() => fetchCourses()`.

This is because, by default, an `Event` object is passed to the function.

Putting `() => fetchCourses()` says to whomever reads this code that `fetchCourses` has no arguments and you don't need further information to do the fetching (no passing of the `Event`).